### ISSUE TITLE
cannot move out of `__token.0` which is behind a shared reference

### ISSUE THREAD

**Issue by Dreamykass** [lalrpop/lalrpop#609]
TL;DR: I have an `Identifier(String)`token enum variant, and it looks like lalrpop doesn't like it.

# The issue:
I wanted to use logos to lex a language, and then parse it with lalrpop. I have a Token enum, and the generated lalrpop parser is supposed to consume these tokens.

The Token enum has a variant like `Identifier(String)`, and it appears that lalrpop generates grammar.rs that fails to compile, with the error as below. And I don't know how to fix it.

There's another project that I looked at, where a String enum variant is present and it compiles fine and works.
Its token enum: <https://github.com/yeslogic/fathom/blob/main/fathom/src/lang/core/lexer.rs#L11>, and its grammar: <https://github.com/yeslogic/fathom/blob/main/fathom/src/lang/core/grammar.lalrpop#L19>

# The full error:
```rs
error[E0507]: cannot move out of `__token.0` which is behind a shared reference
   --> D:\Git\chilimak\target\debug\build\common-aa7aff129626aad4\out/grammar.rs:181:15
    |
181 |         match *__token {
    |               ^^^^^^^^ help: consider borrowing here: `&*__token`
...
191 |             Token::Identifier(String) if true => Some(9),
    |                               ------
    |                               |
    |                               data moved here
    |                               move occurs because `String` has type `std::string::String`, which does not implement the `Copy` trait
```

# The code:
- the whole crate/workspace of my project: <https://github.com/Dreamykass/chilimak/tree/master/common>
- my lalrpop grammar: <https://github.com/Dreamykass/chilimak/blob/master/common/src/grammar.lalrpop>
- my token enum that the lalrpop parser is supposed to consume: <https://github.com/Dreamykass/chilimak/blob/master/common/src/lexer/token.rs>

I tried to change `Identifier => Token::Identifier(<String>)` into `=> Token::Identifier(String)`, or `=> Token::Identifier(<ref String>)`, or similar, but it appears to have no effect.

**Comment by Dreamykass** [comment]
...I was editing the wrong `grammar.lalrpop`, since I had two in my project, and I don't even remember why I created the second one

It does work when I do `Identifier => Token::Identifier(<String>)`, of course
